src/addElasticsearchNode.ts

```diff
@@ -1,6 +1,61 @@
-export async function handler(event: ClusterStatusResponse): Promise<AddElasticsearchNodeResponse> {
+import {ssmCommand} from './utils/ssmCommand';
+import {increaseAsgSize, describeAsg} from './utils/autoscaling';
+import {StandardOutputContent} from 'aws-sdk/clients/ssm';
+import {AutoScalingGroupsType} from "aws-sdk/clients/autoscaling";
+
+export async function handler(event): Promise<AddElasticsearchNodeResponse> {
+
     return new Promise<AddElasticsearchNodeResponse>((resolve, reject) => {
-        const placeholderResponse: AddElasticsearchNodeResponse = { "instanceId": "id123", "expectedClusterSize": 6 };
-        resolve(placeholderResponse);
-    })
+
+        const instanceId: string = event.instanceId;
+        const asg: string = process.env.ASG_NAME;
+
+        const currentCapacity: Promise<number> = describeAsg(asg)
+            .then(asgInfo => handleAsgResponse(asgInfo));
```

Comment:
I'd rename `handleAsgResponse` to `getAsgDesiredCapacity` or something.
Also, it can be shortened to `describeAsg(asg).then(getAsgDesiredCapacity)`